commons-packet/commons-packet-service/src/main/java/io/mosip/commons/packetmanager/service/PacketWriterService.java

```diff
@@ -0,0 +1,147 @@
+package io.mosip.commons.packetmanager.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.mosip.commons.packet.constants.PacketUtilityErrorCodes;
+import io.mosip.commons.packet.dto.TagDeleteResponseDto;
+import io.mosip.commons.packet.dto.TagDto;
+import io.mosip.commons.packet.dto.TagRequestDto;
+import io.mosip.commons.packet.dto.TagResponseDto;
+import io.mosip.commons.packet.exception.TagCreationException;
+import io.mosip.commons.packet.exception.TagDeletionException;
+import io.mosip.commons.packet.facade.PacketReader;
+import io.mosip.commons.packet.facade.PacketWriter;
+import io.mosip.commons.packet.util.PacketManagerLogger;
+import io.mosip.kernel.core.exception.BaseCheckedException;
+import io.mosip.kernel.core.exception.BaseUncheckedException;
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+
+@Component
+public class PacketWriterService {
+    private static Logger LOGGER = PacketManagerLogger.getLogger(PacketWriterService.class);
+    
+    @Autowired
+    private PacketReader packetReader;
+    
+    @Autowired
+    private PacketWriter packetWriter;
+    
+    
+    public TagResponseDto addTags(TagDto tagDto) {
+    	try {
+
+			Map<String, String> existingTags = packetReader.getTags(tagDto.getId());
+				for (Entry<String, String> entry : tagDto.getTags().entrySet()) {
+					if (existingTags.containsKey(entry.getKey())) {
+
+						throw new TagCreationException(PacketUtilityErrorCodes.TAG_ALREADY_EXIST.getErrorCode(),
+								PacketUtilityErrorCodes.TAG_ALREADY_EXIST.getErrorMessage());
+				}
+				}
+
+			tagDto.setTags(tagDto.getTags());

```

Comment:
Can we remove this line as well, since it is getting the tags from same dto and setting it to same dto

Reply:
Fixed